core/FObject.js

```diff
@@ -657,15 +657,28 @@ var FObject = {
   /** Create a deep copy of this object. **/
   deepClone: function() {
     var m = {};
+    var transients = {};
     for ( var key in this.instance_ ) {
       var value = this[key];
       if ( value !== undefined ) {
         var prop = this.model_.getProperty(key);
-        if ( prop && prop.deepCloneProperty )
-          m[key] = prop.deepCloneProperty.call(prop, value);
+        if ( prop && prop.deepCloneProperty ) {
+          if ( prop.transient )
```

Comment:
If we do go this route, we should move this prop.transient check into deepCloneProperty.  We're already letting the property decide how to deep clone itself, it shiould be in charge of this sort of decision.